src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -94,8 +94,8 @@ public Pipe(PipeOptions options)
             _minimumSegmentSize = options.MinimumSegmentSize;
             _pauseWriterThreshold = options.PauseWriterThreshold;
             _resumeWriterThreshold = options.ResumeWriterThreshold;
-            _readerScheduler = options.ReaderScheduler ?? PipeScheduler.Inline;
-            _writerScheduler = options.WriterScheduler ?? PipeScheduler.Inline;
+            _readerScheduler = options.ReaderScheduler ?? PipeScheduler.ThreadPool;
```

Comment:
We either need to move it to PipeOptions or move defaults for memory pool here, it's a bit all over the place right now.